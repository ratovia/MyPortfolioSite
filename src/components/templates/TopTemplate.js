import React from "react"
import styled from "styled-components"
import { FrederickaH1 } from "../atoms/FrederickaText"
import { Layout } from "../layout"
const Header = styled.header`
  width: 400px;
  z-index: 20;
  position: absolute;
  top: 0;
  left: 0;
`

const Nav = styled.div`
  width: 108px;
  height: 100vh;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 10;
`

const Container = styled.div`
  display: flex;
  margin-left: 108px;
  width: calc(100vw - 108px);
  height: 100vh;
`

const LeftContent = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 50%;
  height: 100%;
  background-color: #f04949;
  padding-top: 100px;
`

const RightContent = styled.div`
  display: flex;
  width: 50%;
  height: 100%;
  background-color: #414141;
`

const MainContent = styled.div`
  display: flex;
  margin-left: 108px;
  width: calc(100vw - 108px);
  min-height: 100px;
  background-color: #414141;
`

export const TopTemplate = ({
  headerText,
  navigation,
  leftContent,
  rightContent,
  mainContent,
}) => {
  return (
    <Layout>
      <Header>
        <FrederickaH1 label={headerText} color="white"></FrederickaH1>
      </Header>
      <Nav>{navigation}</Nav>
      <Container>
        <LeftContent>{leftContent}</LeftContent>
        <RightContent>{rightContent}</RightContent>
      </Container>
      <MainContent>{mainContent}</MainContent>
    </Layout>
  )
}
