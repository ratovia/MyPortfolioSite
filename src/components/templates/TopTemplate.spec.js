import React from "react"
import renderer from "react-test-renderer"
import { TopTemplate } from "./TopTemplate"
describe("TopTemplate", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <TopTemplate
          headerText="Hello World"
          navigation={
            <div
              style={{
                width: "100%",
                height: "100%",
                backgroundColor: "red",
              }}
            ></div>
          }
          leftContent={
            <div
              style={{
                width: "100%",
                height: "100%",
                backgroundColor: "blue",
              }}
            ></div>
          }
          rightContent={
            <div
              style={{
                width: "100%",
                height: "100%",
                backgroundColor: "green",
              }}
            ></div>
          }
          mainContent={
            <div
              style={{
                width: "100%",
                minHeight: "100px",
                backgroundColor: "yellow",
              }}
            ></div>
          }
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
