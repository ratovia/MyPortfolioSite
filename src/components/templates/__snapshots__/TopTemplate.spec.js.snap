// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTemplate renders correctly 1`] = `
<div>
  <header
    className="sc-eCImvq hMdXFA"
  >
    <h1
      className="sc-gsDJrp duPHPt"
    >
      <p
        className="sc-bdvvaa iFygtI"
        color="white"
      >
        Hello World
      </p>
    </h1>
  </header>
  <div
    className="sc-jRQAMF kVxEBP"
  >
    <div
      style={
        Object {
          "backgroundColor": "red",
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="sc-gKckTs heCNMp"
  >
    <div
      className="sc-iCfLBT okQih"
    >
      <div
        style={
          Object {
            "backgroundColor": "blue",
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </div>
    <div
      className="sc-furvIG jHKGbJ"
    >
      <div
        style={
          Object {
            "backgroundColor": "green",
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <div
    className="sc-pVTma dplchq"
  >
    <div
      style={
        Object {
          "backgroundColor": "yellow",
          "minHeight": "100px",
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
